### PR TITLE
Add job logging functionalities

### PR DESCRIPTION
**Pull request by khalifapro**

### CODE REVIEW

File: ingestjob/logger.go [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+const (
+	logIndex   = "insights-job-logging"
+	inProgress = "inprogress"
+	failed     = "failed"
+	done       = "done"
+)
+
+// ESLogProvider used in connecting to ES logging server
+type ESLogProvider interface {
+	CreateDocument(index, documentID string, body []byte) ([]byte, error)
```
Comment (linonymous) [inline ingestjob/logger.go:20]: can we also write a function to generate this documentID? This function would be common for all datasource connectors.

Comment (linonymous) [reply ingestjob/logger.go:20]: yep, lets add that as a part of another pull request.

**Review by linonymous — APPROVED** [review]

File: ingestjob/logger.go [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+	doc := map[string]interface{}{
+		"datasource": log.Datasource,
+		"endpoint":   log.Endpoint,
+		"created_at": log.CreatedAt,
+		"status":     log.Status,
+	}
+
+	_, err := s.esClient.UpdateDocument(index, docID, doc)
+	if err != nil {
+		fmt.Println(index)
```
Comment (ajinkyan83) [inline ingestjob/logger.go:162]: We can remove this debug statement

File: ingestjob/logger.go [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+	"fmt"
+	"time"
+
+	"github.com/LF-Engineering/insights-datasource-shared/uuid"
+	jsoniter "github.com/json-iterator/go"
+)
+
+const (
+	logIndex   = "insights-job-logging"
+	inProgress = "inprogress"
```
Comment (ajinkyan83) [inline ingestjob/logger.go:13]: Do we need to export these status constants for use in Data sources?

**Review by lukaszgryglicki — APPROVED** [review]
/lgtm

**Review by ajinkyan83 — APPROVED** [review]

**Review by fayazg — CHANGES_REQUESTED** [review]
@enkhalifapro @linonymous we want to avoid having the client figure out what to do with document ID or what the index should be. Also, `ProjectSlug` is not needed sine we have a `Configuration` array.
Index and DocumentID should be kept internal to this function, documentID can be hash of `Connector`+`Configuration`, which should keep it unique so that it overwrites the same document when `Status` changes.
And index can be `insights-connector-{Connector}-log`, 1 index per connector.

File: ingestjob/logger.go (hunk 1 above)
Comment (khalifapro) [reply ingestjob/logger.go:22]: No need for having it in a separate func cuz it's used only within write func

**Review by fayazg — APPROVED** [review]

**Pull request merged**